### PR TITLE
stats: Improve test coverage

### PR DESCRIPTION
**Pull request by jmarantz**
Description: a few methods here and there that were untested. Either remove them or add unit tests.
Risk Level: low
Testing: //test/common/stats/...
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/stats/allocator_impl.h [hunk 1]
```diff
@@ -46,8 +46,13 @@ class AllocatorImpl : public Allocator {
     bool operator()(const Metric* a, const Metric* b) const {
       return a->statName() == b->statName();
     }
-    bool operator()(StatName a, const Metric* b) const { return a == b->statName(); }
     bool operator()(const Metric* a, StatName b) const { return a->statName() == b; }
+    // This comparison pattern does not currently occur due to the implementation of
```
Comment (mattklein123) [inline source/common/stats/allocator_impl.h]: WDYT about just deleting this for now and we can fix the compiler error later?

Comment (lizan) [reply source/common/stats/allocator_impl.h]: +1, or just leave it

Comment (jmarantz) [reply source/common/stats/allocator_impl.h]: I don't have a strong opinion; but are you asking to delete the entire comment? Or just the commented out method?

Comment (mattklein123) [reply source/common/stats/allocator_impl.h]: I would just delete all of it personally.

**Pull request merged**